### PR TITLE
[FIX] pos_restaurant: transferring order to a table already containing order

### PR DESCRIPTION
**Pull request by dka-odoo**
Issue :- https://www.odoo.com/web?#id=656927&view_type=form&model=project.issue&menu_id=4120&action=609

### CODE REVIEW

File: addons/pos_restaurant/static/src/js/floors.js [hunk 1]
```diff
@@ -784,13 +784,12 @@ models.PosModel = models.PosModel.extend({
             this.set_order(null);
         } else if (this.order_to_transfer_to_different_table) {
             this.order_to_transfer_to_different_table.table = table;
+            this.db.save_unpaid_order(this.order_to_transfer_to_different_table);
```
Comment (jorenvo) [inline addons/pos_restaurant/static/src/js/floors.js]: Call `save_to_db()` of `this.order_to_transfer_to_different_table` instead. That's what we use in the rest of the POS. By directly calling `save_unpaid_order()` you bypass the posmodel `init_locked` lock.

**Comment by jorenvo** [comment]
The commit message should be changed to something more descriptive. It should precisely describe the bug that occurs, not just "issue, when transferring order from one table to another". For one, the bug only occurs when there is already an order on the table you transfer it to. Also, it's always nice if the commit message explains why exactly something wasn't working, and why your patch fixes it.

So something like:

```
[FIX] pos_restaurant: transferring order to a table already containing order

Before this, transferring an order to a table which already contains an
order triggered the change on the order already on the table. Not the newly
transferred order. This fixes that by instead explicitly calling
save_to_db() on the transferred order instead.
```

would be a lot better.

File: addons/pos_restaurant/static/src/js/floors.js (hunk 1 above)
Comment (dka-odoo) [inline addons/pos_restaurant/static/src/js/floors.js]: @jorenvo  done!  thanks :)

**Comment by jorenvo** [comment]
Thanks, merged in saas-7 at dad1e378258894f0d77fa56c398d809de6ac8162.

**Pull request closed without merging**